guides/ci/travis.md

```diff
@@ -0,0 +1,358 @@
+Here is the example:
+
+```yaml
+# .travis.yml
+matrix:
+ include:
+   - language: crystal
+     crystal:
+       - latest
+     script:
+       - crystal spec
+
+   - language: crystal
+     crystal:
+       - nightly
+     script:
+       - crystal spec
+
+   - language: bash
+     services:
+       - docker
+     script:
+       - docker run -v $PWD:/src -w /src crystallang/crystal:0.31.1 crystal spec
+```
+
+## Using external dependencies
+
+Before presenting some examples, it is important to mention that there are many ways to achieve this and **it will heavily depend on our development workflow**.
+
+With this in mind, let's continue!
+
+Using shards dependencies (i.e. libraries declared in `shard.yml`) will not be a problem, because thanks to the `language: crystal` support, TravisCI automatically runs `shards install`. But, how do we install dependencies outside `shards.yml`? Well, here is a first example installing SQLite3 using Travis CI configuration file:
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+before_install:
+  - sudo apt-get -y install sqlite3
+
+addons:
+  apt:
+    update: true
+
+script:
+  - sqlite3 --version
+  - crystal spec
+```
+
+**Note:** We are using `sqlite3` just for printing SQLite's version, as an example of external dependency.
+
+The same can be accomplished using a containerization service like Docker. Here is an example of a `Dockerfile` installing SQLite3:
+
+```dockerfile
+# Dockerfile
+FROM crystallang/crystal:latest
+
+RUN apt-get update && apt-get install -y sqlite3
+```
+
+And here is the TravisCI configuration file:
+
+```yml
+# .travis.yml
+language: bash
+
+services:
+  - docker
+
+before_install:
+  # build image using Dockerfile:
+  - docker build -t testing .
+
+script:
+  # run specs in the container
+  - docker run -v $PWD:/src -w /src testing crystal spec
+  # sqlite3
+  - docker run -v $PWD:/src -w /src testing sqlite3 --version
+```
+
+Again, these are only examples of two ways of running our application and installing (and using) dependencies.
+
+> *Note:* Dockerfile arguments can be used to use the same Dockerfile for latest, nightly or a specific version.
+
+## Using a database (example using MySQL)
+
+As you may notice, in the last configuration file we are using `docker` as a `service`. Travis CI may start different services as needed. For example, we may need [MySQL](https://docs.travis-ci.com/user/database-setup/#mysql).
+
+```yaml
+# .travis.yml
+language: crystal
+crystal:
+  - latest
+
+services:
+  - mysql
+
+script:
+  - crystal spec
+```
+
+To continue this example, let's add a new test that uses the database.
+(**important:** the new test case is only for testing Travis CI database service, and it's not a good design or test case by any means)
+
+```crystal
+# game_of_life_spec.cr
```

Comment:
This example is way to complex for a simple demonstration.
Just putting `DB.connect` in a spec file is completely sufficient.

Reply:
Yep, it is a little more complex example than `DB.connect` but it has another purpose: it shows how to use `environment variables` (in this case `DATABASE_URL`)
For making the example shorter, now we only show the lines that need to be added to the example introduced in Continuous Integration.

Reply:
Really, I would just make the example be as tight as possible.
```cr
require "mysql"
require "spec"

it "connects do database" do
  DB.connect(ENV["DATABASE_URL"]) do |db|
    db.query("SELECT 'foo'", as: String). should eq "foo"
  end
end
```

There's no point in having anything else here. It's just noise.

Reply:
oh! ok, that's a simpler example, but I forgot to mention that the original example has another goal: to show how to use a `setup.sql` file to initialize the database from TravisCI. This may seem trivial but it's not (that's why I even mention the source of the solution: https://andidittrich.de/2017/06/travisci-setup-mysql-tablesdata-before-running-tests.html)
So we wanted to create an example that the reader could almost cut&paste.

I think that, after merging this new section, we may iterate with a new and more simple example (oh! and we would update the example in the introduction as well!) and then create a new PR with those changes. 

@straight-shoota, @bcardiff:  Does it seem ok?

By the way: thank you for all the suggestions!!

Reply:
I seriously doubt that *setting up* a MySQL database fits in the scope of this guide. The goal of this section should be to demonstrate the concept of using services on travis.
How to actually use such a service is a highly specific topic. MySQL is really just a stand in for any travis service. You wouldn't want to dive into giving instructions for other services either.
It should be sufficient to refer to https://docs.travis-ci.com/user/database-setup/ for more specific documentation.

Reply:
ok! I removed the current example and added the simpler DB example.

Let me know what you think!